primitives/src/signatures/mod.rs

```diff
@@ -91,14 +91,39 @@ pub trait SignatureScheme {
 }
 
 /// Trait for aggregatable signatures.
-pub trait AggregateableSignatureSchemes<H>: SignatureScheme {
-    // TODO: APIs for aggregateable signatures
+/// TODO: generic over hash functions
+pub trait AggregateableSignatureSchemes: SignatureScheme {
+    /// Aggregate multiple signatures into a single signature
+    fn aggregate(
+        pp: &Self::PublicParameter,
+        sigs: &[Self::Signature],
+    ) -> Result<Self::Signature, PrimitivesError>;
```

Comment:
I'm not sure why `aggregate()` doesn't take in `{pk_i}` as input.

WDYT of the interfaces presented [here](https://docs.google.com/presentation/d/1G4XGqrBLwqMyDQce_xpPQUEMOK4lFrneuvGYU3MVDsI/edit?usp=sharing)? @chancharles92

Reply:
We are following the IETF standard here: https://www.ietf.org/archive/id/draft-irtf-cfrg-bls-signature-05.html#name-aggregate

Reply:
Also in our setting, the key aggregation should be done by the verifier. There is no benefit for the signer to do key aggregation.

Reply:
Ah I guess Schnorr/ECDSA might need `pk_i` as inputs.